core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2626,8 +2601,12 @@ class ReplicaManager(val config: KafkaConfig,
     localLeaders.forKeyValue { (tp, info) =>
       getOrCreatePartition(tp, delta, info.topicId).foreach { case (partition, isNew) =>
         try {
+          val partitionDirectoryId: Option[Uuid] =
+            Option(newImage.topics().getPartition(info.topicId(), partition.partitionId)).map(_.directory(localBrokerId))
```

Comment:
Maybe we can default this to `MIGRATING` instead of an `Option`?

Reply:
Do we need to count `MIGRATING` now? what would this mean for creating the directory?